docs/proposals/push-based-metrics-collection.md

```diff
@@ -0,0 +1,195 @@
+
+![](../images/push-based-metrics-collection.png)
+
+Fig.1 Architecture of the new design
+
+### Goals
+1. **A new parameter in Python SDK function `tune`**: allow users to specify the method of collecting metrics(push-based/pull-based).
+
+2. **A new interface `report_metrics` in Python SDK**: push the metrics to Katib DB directly.
+
+3. The final metrics of worker pods should be **pushed to Katib DB directly** in the push mode of metrics collection.
+
+### Non-Goals
+1. Implement authentication model for Katib DB to push metrics.
+
+2. Support pushing data to different types of storage system(prometheus, self-defined interface etc.)
+
+
+## API
+
+### New Parameter in Python SDK Function `tune`
+
+We decided to add `metrics_collection_mechanism` to `tune` function in Python SDK.
+
+```Python
+def tune(
+    self,
+    name: str,
+    objective: Callable,
+    parameters: Dict[str, Any],
+    base_image: str = constants.BASE_IMAGE_TENSORFLOW,
+    namespace: Optional[str] = None,
+    env_per_trial: Optional[Union[Dict[str, str], List[Union[client.V1EnvVar, client.V1EnvFromSource]]]] = None,
+    algorithm_name: str = "random",
+    algorithm_settings: Union[dict, List[models.V1beta1AlgorithmSetting], None] = None,
+    objective_metric_name: str = None,
+    additional_metric_names: List[str] = [],
+    objective_type: str = "maximize",
+    objective_goal: float = None,
+    max_trial_count: int = None,
+    parallel_trial_count: int = None,
+    max_failed_trial_count: int = None,
+    resources_per_trial: Union[dict, client.V1ResourceRequirements, None] = None,
+    retain_trials: bool = False,
+    packages_to_install: List[str] = None,
+    pip_index_url: str = "https://pypi.org/simple",
+    # The newly added parameter metrics_collector_config.
+    # It specifies the config of metrics collector, for example, 
+    # metrics_collector_config={"kind": "Push"},
+    metrics_collector_config: Dict[str, Any] = None, 
+)
+```
+
+### New Interface `report_metrics` in Python SDK
+
+```Python
+"""Push Metrics Directly to Katib DB
+    Katib DB Manager service should be accessible while calling this API.
+
+    If you run this API in-cluster (e.g. from the Kubeflow Notebook) you can
+    use the default Katib DB Manager address: `katib-db-manager.kubeflow:6789`.
+
+    If you run this API outside the cluster, you have to port-forward the
+    Katib DB Manager before getting the Trial metrics: `kubectl port-forward svc/katib-db-manager -n kubeflow 6789`.
+    In that case, you can use this Katib DB Manager address: `localhost:6789`.
+
+    You can use `curl` to verify that Katib DB Manager is reachable: `curl <db-manager-address>`.
+
+    [!!!] Trial's namespace and name should always be passed into Katib Trials as env variable `KATIB_TRIAL_NAMESPACE` and `KATIB_TRIAL_NAME`.
```

Comment:
I think, we don't need to pass `KATIB_TRIAL_NAMESPACE` to the Trial since can always get the container namespace from `/var/run/secrets/kubernetes.io/serviceaccount/namespace`.
WDYT @Electronic-Waste @tenzen-y @johnugeorge ?

Reply:
I agree with you Andrey.